### PR TITLE
[quant][core][gpu][feature] Implemented quantized cuda gelu

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77212

Summary:
Support for quantized cuda gelu has been provided by using
`dequantize -> fp32 cuda gelu kernel -> quantize`. Mathematically, this
is not equivalent to doing int8 gelu, so we have opted for this approach
for now. It might be possible to write a variant of the int8 gelu that's
equivalent to `dequantize -> fp32 cuda gelu kernel -> quantize`, which
can be a topic for future work.

Test function `test_qgelu` was amended to test gelu for quantized cuda
backends.

Test Plan:
```
python test/test_quantization.py -k test_qgelu
```

Differential Revision: [D36392774](https://our.internmc.facebook.com/intern/diff/D36392774)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36302475).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36302475).

**Comment by dzdang** [comment]
@jerryzh168 wondering if we should move `Activation.cpp` in `quantized/cuda` instead since currently it has nothing to do with `cuDNN`?

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36302475).

File: aten/src/ATen/native/quantized/cuda/Activation.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// this kernel is currently implemented with dequantize -> fp32 gelu -> quantize, which is not equivalent to int8 gelu
+// It might be possible to write a variant of the int8 gelu that's equivalent to dequantize -> fp32 cuda gelu kernel -> quantize,
+// which can be a topic for future work.
+Tensor gelu_quantized_cuda(const Tensor& qx, c10::string_view approximate) {
+  if (qx.numel() == 0) {
+    return Tensor{};
+  }
+  auto x_fp32 = at::dequantize(qx);
+  auto result_fp32 = at::gelu(x_fp32);
+  return at::quantize_per_tensor(result_fp32, qx.q_scale(), qx.q_zero_point(), qx.scalar_type());
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cuda/Activation.cpp:16]: if each one of them supports L11-13 we can remove L11-13 right? should we add a TODO to do this?

**Review by jerryzh168 — APPROVED** [review]

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36302475).

**Comment by malfet** [comment]
Reverting this PR internally as it broke bazel builds, see https://hud.pytorch.org/pytorch/pytorch/commit/b892b85b881c7b3b2b6bde529c4d174e348ba9fb

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36392774).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36392774).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36392774).

**Pull request closed without merging**